lib/realtime/tenants/repo/migrations/20240917170412_add_payload_to_messages.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Realtime.Tenants.Migrations.AddPayloadToMessages do
+  @moduledoc false
+  use Ecto.Migration
+
+  def change do
+    alter table(:messages) do
+      add_if_not_exists :payload, :map
+      add_if_not_exists :event, :text
+      add_if_not_exists :topic, :text
+      modify :inserted_at, :utc_datetime, default: fragment("now()")
+      modify :updated_at, :utc_datetime, default: fragment("now()")
+    end
+
+    execute """
+    CREATE OR REPLACE FUNCTION realtime.send(payload jsonb, event text, topic text, extension text DEFAULT 'broadcast')
+    RETURNS void
+    AS $$
+    BEGIN
+        INSERT INTO realtime.messages (payload, event, topic, extension)
+        VALUES (payload, event, topic, extension);
+    END;
+    $$
+    LANGUAGE plpgsql;
+    """
+
+    execute """
+    CREATE OR REPLACE FUNCTION realtime.broadcast ()
```

Comment:
Should this fn maybe be `realtime.broadcast_change`?

Reply:
will change 👍 otherwise it might be too abstract

Reply:
I like realtime.broadcast bc then I can use it via Postgrest or something to broadcast anything. realtime.broadcast_change seems very specific

Reply:
This is a trigger function which specifically broadcasts changes from a table tho.

Reply:
that would be the `send` function we have above. we could rename it to be `broadcast` and that one indeed can send anything detached from pg operations

Reply:
Use `realtime.send` anywhere:

```
SELECT realtime.send(
    '{"message": "Hello, world!"}'::jsonb, -- payload
    'new_message',                         -- event
    'some_room_id',                        -- topic
    'broadcast'                            -- extension (optional, defaults to 'broadcast')
);
```

Reply:
Or both are called realtime.broadcast with different signatures and that way devs don't have to remember two different function names

Reply:
@filipecabaco btw nit but I would do the args like: 

```
SELECT realtime.send(
    'some_room_id',                                         -- topic
    'new_message',                                          -- event
    '{"message": "Hello, world!"}'::jsonb,      -- payload (optional, defaults to '{}')
    'broadcast',                                                 -- extension (optional, defaults to 'broadcast')
    true                                                              -- private (optional, defaults to true)
);
```

Reply:
> Or both are called realtime.broadcast with different signatures and that way devs don't have to remember two different function names

Even more confusing imo.